src/muted.rs

```diff
@@ -0,0 +1,53 @@
+use nostrdb::Note;
+use std::collections::BTreeSet;
+
+use tracing::debug;
+
+#[derive(Default)]
+pub struct Muted {
+    // TODO - implement private mutes
+    pub pubkeys: BTreeSet<[u8; 32]>,
+    pub hashtags: BTreeSet<String>,
+    pub words: BTreeSet<String>,
+    pub threads: BTreeSet<[u8; 32]>,
+}
+
+impl std::fmt::Debug for Muted {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Muted")
+            .field(
+                "pubkeys",
+                &self.pubkeys.iter().map(hex::encode).collect::<Vec<_>>(),
+            )
+            .field("hashtags", &self.hashtags)
+            .field("words", &self.words)
+            .field(
+                "threads",
+                &self.threads.iter().map(hex::encode).collect::<Vec<_>>(),
+            )
+            .finish()
+    }
+}
+
+impl Muted {
+    pub fn is_muted(&self, note: &Note) -> bool {
+        if self.pubkeys.contains(note.pubkey()) {
+            debug!(
+                "{}: MUTED pubkey: {}",
+                hex::encode(note.id()),
+                hex::encode(note.pubkey())
+            );
+            return true;
```

Comment:
we probably don't want to hex encode for every is_muted check. these will be evaluated regardless of the log being enabled.